cgroups/root.go

```diff
@@ -31,35 +30,35 @@ func Setup() error {
 	}
 
 	// Enable controllers we need
-	subtree_control := rootPath + "/cgroup.subtree_control"
-	if err := os.WriteFile(subtree_control, []byte("+memory +io +cpu"), 0644); err != nil {
+	subtreeControl := rootPath + "/cgroup.subtree_control"
+	if err := os.WriteFile(subtreeControl, []byte("+memory +io +cpu"), 0644); err != nil {
 		return fmt.Errorf("failed to enable cgroup controllers for the root cgroup: %w", err)
 	}
 
 	return nil
 }
 
+// TearDown cleans up our cgroups before process shutdown
+func TearDown() error {
+	log.Println("Cleaning up cgroups...")
+	return os.RemoveAll(rootPath)
+}
+
 //-------------------------------------------------------------------------------------------------
 func cgroup2Mounted() (bool, error) {
-	mounts_file, err := os.Open("/proc/mounts")
+	mountsFile, err := os.Open("/proc/mounts")
 	if err != nil {
 		return false, fmt.Errorf("failed to open the mounts list: %w", err)
 	}
-	defer mounts_file.Close()
 
-	scanner := bufio.NewScanner(mounts_file)
+	scanner := bufio.NewScanner(mountsFile)
 	for scanner.Scan() {
 		fields := strings.Split(scanner.Text(), " ")
```

Comment:
`strings.SplitN` can limit the number of tokens to 2 (as you only need the first one).